### ISSUE TITLE
Frameworks with Objective-C++ producing missing symbols

### ISSUE THREAD

**Issue by HeEAaD** [Ruenzuo/cocoapods-amimono#8]
# Issue
Patching Pods with Amimono which are using Objective-C++ will result in missing symbols during the linker phase.

# Steps to reproduce
1. add Amimono and a Pod with Objective-C++ in the Podfile (see the example project: [Cpp Example.zip](https://github.com/Ruenzuo/cocoapods-amimono/files/422977/Cpp.Example.zip) )
2. pod install
3. open the generated workspace
4. build the project
5. You will see missing symbols errors during the linker phase

# Output from the "Cpp Example"
```
Ld /Users/steffen.matthischke/Library/Developer/Xcode/DerivedData/Cpp_Example-fymgldifskmkengtenoxhvgbekdp/Build/Products/Debug-iphonesimulator/Cpp\ Example.app/Cpp\ Example normal x86_64
    cd "/Users/steffen.matthischke/CODING/cocoapods-amimono-example/Cpp Example"
    export IPHONEOS_DEPLOYMENT_TARGET=9.3
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk -L/Users/steffen.matthischke/Library/Developer/Xcode/DerivedData/Cpp_Example-fymgldifskmkengtenoxhvgbekdp/Build/Products/Debug-iphonesimulator -F/Users/steffen.matthischke/Library/Developer/Xcode/DerivedData/Cpp_Example-fymgldifskmkengtenoxhvgbekdp/Build/Products/Debug-iphonesimulator -F/Users/steffen.matthischke/Library/Developer/Xcode/DerivedData/Cpp_Example-fymgldifskmkengtenoxhvgbekdp/Build/Products/Debug-iphonesimulator/pop -filelist /Users/steffen.matthischke/Library/Developer/Xcode/DerivedData/Cpp_Example-fymgldifskmkengtenoxhvgbekdp/Build/Intermediates/Cpp\ Example.build/Debug-iphonesimulator/Cpp\ Example.build/Objects-normal/x86_64/Cpp\ Example.LinkFileList -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -Xlinker -rpath -Xlinker @executable_path/Frameworks -mios-simulator-version-min=9.3 -Xlinker -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -filelist /Users/steffen.matthischke/Library/Developer/Xcode/DerivedData/Cpp_Example-fymgldifskmkengtenoxhvgbekdp/Build/Intermediates/Pods.build/Debug-iphonesimulator-x86_64.objects.filelist -fobjc-arc -fobjc-link-runtime -framework Pods_Cpp_Example -Xlinker -dependency_info -Xlinker /Users/steffen.matthischke/Library/Developer/Xcode/DerivedData/Cpp_Example-fymgldifskmkengtenoxhvgbekdp/Build/Intermediates/Cpp\ Example.build/Debug-iphonesimulator/Cpp\ Example.build/Objects-normal/x86_64/Cpp\ Example_dependency_info.dat -o /Users/steffen.matthischke/Library/Developer/Xcode/DerivedData/Cpp_Example-fymgldifskmkengtenoxhvgbekdp/Build/Products/Debug-iphonesimulator/Cpp\ Example.app/Cpp\ Example

Undefined symbols for architecture x86_64:
  "std::__1::__vector_base_common<true>::__throw_length_error() const", referenced from:
      std::__1::vector<std::__1::shared_ptr<POPAnimatorItem>, std::__1::allocator<std::__1::shared_ptr<POPAnimatorItem> > >::allocate(unsigned long) in POPAnimator.o
  "std::__1::__shared_weak_count::__add_shared()", referenced from:
      -[POPAnimator _renderTime:items:] in POPAnimator.o
      -[POPAnimator _renderTime:item:] in POPAnimator.o
      -[POPAnimator removeAllAnimationsForObject:] in POPAnimator.o
      -[POPAnimator removeAnimationForObject:key:cleanupDict:] in POPAnimator.o
      updateAnimatable(objc_object*, _POPPropertyAnimationState*, bool) in POPAnimator.o
      std::__1::list<std::__1::shared_ptr<POPAnimatorItem>, std::__1::allocator<std::__1::shared_ptr<POPAnimatorItem> > >::push_back(std::__1::shared_ptr<POPAnimatorItem> const&) in POPAnimator.o
      _POPPropertyAnimationState::finalizeProgress() in POPAnimator.o
      ...
  "std::__1::__shared_weak_count::__release_shared()", referenced from:
      std::__1::shared_ptr<POP::Vector>::~shared_ptr() in POPAnimationRuntime.o
      std::__1::shared_ptr<POP::Vector const>::~shared_ptr() in POPAnimator.o
      std::__1::shared_ptr<POP::Vector>::~shared_ptr() in POPAnimator.o
      std::__1::shared_ptr<POPAnimatorItem>::~shared_ptr() in POPAnimator.o
      std::__1::shared_ptr<POP::Vector>::~shared_ptr() in POPBasicAnimation.o
      std::__1::shared_ptr<POP::Vector const>::~shared_ptr() in POPBasicAnimation.o
      std::__1::shared_ptr<POP::Vector>::~shared_ptr() in POPDecayAnimation.o
      ...
  "std::__1::__shared_weak_count::~__shared_weak_count()", referenced from:
      std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> >::~__shared_ptr_pointer() in POPAnimationRuntime.o
      std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> >::~__shared_ptr_pointer() in POPAnimator.o
      std::__1::__shared_ptr_pointer<POPAnimatorItem*, std::__1::default_delete<POPAnimatorItem>, std::__1::allocator<POPAnimatorItem> >::~__shared_ptr_pointer() in POPAnimator.o
      std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> >::~__shared_ptr_pointer() in POPBasicAnimation.o
      std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> >::~__shared_ptr_pointer() in POPDecayAnimation.o
      std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> >::~__shared_ptr_pointer() in POPPropertyAnimation.o
      std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> >::~__shared_ptr_pointer() in POPSpringAnimation.o
      ...
  "std::terminate()", referenced from:
      ___clang_call_terminate in POPAnimation.o
      ___clang_call_terminate in POPAnimationRuntime.o
      ___clang_call_terminate in POPAnimator.o
      ___clang_call_terminate in POPBasicAnimation.o
      ___clang_call_terminate in POPDecayAnimation.o
      ___clang_call_terminate in POPPropertyAnimation.o
      ___clang_call_terminate in POPSpringAnimation.o
      ...
  "typeinfo for std::__1::__shared_weak_count", referenced from:
      typeinfo for std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> > in POPAnimationRuntime.o
      typeinfo for std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> > in POPAnimator.o
      typeinfo for std::__1::__shared_ptr_pointer<POPAnimatorItem*, std::__1::default_delete<POPAnimatorItem>, std::__1::allocator<POPAnimatorItem> > in POPAnimator.o
      typeinfo for std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> > in POPBasicAnimation.o
      typeinfo for std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> > in POPDecayAnimation.o
      typeinfo for std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> > in POPPropertyAnimation.o
      typeinfo for std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> > in POPSpringAnimation.o
      ...
  "vtable for __cxxabiv1::__class_type_info", referenced from:
      typeinfo for _POPAnimationState in POPAnimation.o
      typeinfo for std::__1::default_delete<POP::Vector> in POPAnimationRuntime.o
      typeinfo for _POPAnimationState in POPAnimator.o
      typeinfo for std::__1::default_delete<POP::Vector> in POPAnimator.o
      typeinfo for std::__1::default_delete<POPAnimatorItem> in POPAnimator.o
      typeinfo for _POPAnimationState in POPBasicAnimation.o
      typeinfo for std::__1::default_delete<POP::Vector> in POPBasicAnimation.o
      ...
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for __cxxabiv1::__si_class_type_info", referenced from:
      typeinfo for std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> > in POPAnimationRuntime.o
      typeinfo for _POPPropertyAnimationState in POPAnimator.o
      typeinfo for std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> > in POPAnimator.o
      typeinfo for std::__1::__shared_ptr_pointer<POPAnimatorItem*, std::__1::default_delete<POPAnimatorItem>, std::__1::allocator<POPAnimatorItem> > in POPAnimator.o
      typeinfo for _POPPropertyAnimationState in POPBasicAnimation.o
      typeinfo for _POPBasicAnimationState in POPBasicAnimation.o
      typeinfo for std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> > in POPBasicAnimation.o
      ...
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for std::__1::__shared_count", referenced from:
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPAnimationRuntime.o
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPAnimator.o
      std::__1::shared_ptr<POPAnimatorItem>::shared_ptr<POPAnimatorItem>(POPAnimatorItem*, std::__1::enable_if<is_convertible<POPAnimatorItem*, POPAnimatorItem*>::value, std::__1::shared_ptr<POPAnimatorItem>::__nat>::type) in POPAnimator.o
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPBasicAnimation.o
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPDecayAnimation.o
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPPropertyAnimation.o
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPSpringAnimation.o
      ...
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for std::__1::__shared_weak_count", referenced from:
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPAnimationRuntime.o
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPAnimator.o
      std::__1::shared_ptr<POPAnimatorItem>::shared_ptr<POPAnimatorItem>(POPAnimatorItem*, std::__1::enable_if<is_convertible<POPAnimatorItem*, POPAnimatorItem*>::value, std::__1::shared_ptr<POPAnimatorItem>::__nat>::type) in POPAnimator.o
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPBasicAnimation.o
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPDecayAnimation.o
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPPropertyAnimation.o
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPSpringAnimation.o
      ...
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "operator delete(void*)", referenced from:
      -[POPAnimation _initState] in POPAnimation.o
      _POPAnimationState::~_POPAnimationState() in POPAnimation.o
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPAnimationRuntime.o
      std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> >::~__shared_ptr_pointer() in POPAnimationRuntime.o
      std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> >::__on_zero_shared() in POPAnimationRuntime.o
      std::__1::__shared_ptr_pointer<POP::Vector*, std::__1::default_delete<POP::Vector>, std::__1::allocator<POP::Vector> >::__on_zero_shared_weak() in POPAnimationRuntime.o
      std::__1::list<std::__1::shared_ptr<POPAnimatorItem>, std::__1::allocator<std::__1::shared_ptr<POPAnimatorItem> > >::push_back(std::__1::shared_ptr<POPAnimatorItem> const&) in POPAnimator.o
      ...
  "operator new(unsigned long)", referenced from:
      -[POPAnimation _initState] in POPAnimation.o
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPAnimationRuntime.o
      -[POPAnimator addAnimation:forObject:key:] in POPAnimator.o
      std::__1::list<std::__1::shared_ptr<POPAnimatorItem>, std::__1::allocator<std::__1::shared_ptr<POPAnimatorItem> > >::push_back(std::__1::shared_ptr<POPAnimatorItem> const&) in POPAnimator.o
      std::__1::shared_ptr<POP::Vector>::shared_ptr<POP::Vector>(POP::Vector*, std::__1::enable_if<is_convertible<POP::Vector*, POP::Vector*>::value, std::__1::shared_ptr<POP::Vector>::__nat>::type) in POPAnimator.o
      std::__1::vector<std::__1::shared_ptr<POPAnimatorItem>, std::__1::allocator<std::__1::shared_ptr<POPAnimatorItem> > >::allocate(unsigned long) in POPAnimator.o
      std::__1::shared_ptr<POPAnimatorItem>::shared_ptr<POPAnimatorItem>(POPAnimatorItem*, std::__1::enable_if<is_convertible<POPAnimatorItem*, POPAnimatorItem*>::value, std::__1::shared_ptr<POPAnimatorItem>::__nat>::type) in POPAnimator.o
      ...
  "___cxa_begin_catch", referenced from:
      ___clang_call_terminate in POPAnimation.o
      ___clang_call_terminate in POPAnimationRuntime.o
      ___clang_call_terminate in POPAnimator.o
      ___clang_call_terminate in POPBasicAnimation.o
      ___clang_call_terminate in POPDecayAnimation.o
      ___clang_call_terminate in POPPropertyAnimation.o
      ___clang_call_terminate in POPSpringAnimation.o
      ...
  "___dynamic_cast", referenced from:
      applyAnimationTime(objc_object*, _POPAnimationState*, double) in POPAnimator.o
      applyAnimationToValue(objc_object*, _POPAnimationState*) in POPAnimator.o
  "___gxx_personality_v0", referenced from:
      __ZL13_staticStates_block_invoke in POPAnimatableProperty.o
      __ZL13_staticStates_block_invoke_2 in POPAnimatableProperty.o
      __ZL13_staticStates_block_invoke_3 in POPAnimatableProperty.o
      __ZL13_staticStates_block_invoke_4 in POPAnimatableProperty.o
      __ZL13_staticStates_block_invoke_5 in POPAnimatableProperty.o
      __ZL13_staticStates_block_invoke_6 in POPAnimatableProperty.o
      __ZL13_staticStates_block_invoke_7 in POPAnimatableProperty.o
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by Ruenzuo** [comment]
Looks like there are some linker flags and configurations missing in the final `xcconfig` that integrates into the application project, namely:

```
CLANG_CXX_LANGUAGE_STANDARD = c++11
CLANG_CXX_LIBRARY = libc++
OTHER_LDFLAGS = -l"c++"
```

I don't know how CocoaPods handles this for static libraries, but I will have a look. A possible fix would be to make CocoaPods update the `xcconfig` with the missing flags (as it is probably doing for static libraries) like I did for the [resource bundles](https://github.com/Ruenzuo/cocoapods-amimono/blob/master/lib/cocoapods-amimono/patcher.rb#L9-L12). Will have a look at this over the weekend, for now you can add those missing flags into your xcconfigs (all configurations) and that should fix the problem.

**Comment by Ruenzuo** [comment]
Thank you for reporting this issue. It should be fixed on v0.0.3.
